example/zForceKeyboardMouse/zForceKeyboardMouse.ino

```diff
@@ -0,0 +1,214 @@
+
+long globalMillis = millis();          // global timestamp
+const int keyboardBoundary = 1000;     // separate mouse area and keyboard area on the x-axis by 1000 (1/10)mm
+const int holdTime = 80;               // sensitivity for mouse "left click", unit in milli-second
+
+typedef struct Location
+{
+  uint16_t x;
+  uint16_t y;
+} Location;
+
+Location previousLoc;
+
+void setup()
+{
+  Keyboard.begin();           // initialize keyboard
+  Mouse.begin();              // initialize mouse
+  Serial.begin(115200);
+  Serial.println("zforce start");
+  zforce.Start(DATA_READY);
+
+  Message* msg = zforce.GetMessage();
+
+  if (msg != NULL)
+  {
+    Serial.println("Received Boot Complete Notification");
+    Serial.print("Message type is: ");
+    Serial.println((int)msg->type);
+    zforce.DestroyMessage(msg);
+  }
+
+  // Send and read ReverseX
+  zforce.ReverseX(false);
+
+  do
+  {
+    msg = zforce.GetMessage();
+  } while (msg == NULL);
+
+  if (msg->type == MessageType::REVERSEXTYPE)
+  {
+    Serial.println("Received ReverseX Response");
+    Serial.print("Message type is: ");
+    Serial.println((int)msg->type);
+  }
+
+  zforce.DestroyMessage(msg);
+
+
+  // Send and read ReverseY
+  zforce.ReverseY(false);
+
+  do
+  {
+    msg = zforce.GetMessage();
+  } while (msg == NULL);
+
+  if (msg->type == MessageType::REVERSEYTYPE)
+  {
+    Serial.println("Received ReverseY Response");
+    Serial.print("Message type is: ");
+    Serial.println((int)msg->type);
+  }
+
+  zforce.DestroyMessage(msg);
+
+  // Send and read Touch Active Area
+  zforce.TouchActiveArea(0, 0, 4000, 4000);
+
+  do
+  {
+    msg = zforce.GetMessage();
+  } while (msg == NULL);
+
+  if (msg->type == MessageType::TOUCHACTIVEAREATYPE)
+  {
+    Serial.print("minX is: ");
+    Serial.println(((TouchActiveAreaMessage*)msg)->minX);
+    Serial.print("minY is: ");
+    Serial.println(((TouchActiveAreaMessage*)msg)->minY);
+    Serial.print("maxX is: ");
+    Serial.println(((TouchActiveAreaMessage*)msg)->maxX);
+    Serial.print("maxY is: ");
+    Serial.println(((TouchActiveAreaMessage*)msg)->maxY);
+  }
+
+  zforce.DestroyMessage(msg);
+
+  // Send and read Enable
+
+  zforce.Enable(true);
+
+  msg = zforce.GetMessage();
+
+  do
+  {
+    msg = zforce.GetMessage();
+  } while (msg == NULL);
+
+  if (msg->type == MessageType::ENABLETYPE)
+  {
+    Serial.print("Message type is: ");
+    Serial.println((int)msg->type);
+    Serial.println("Sensor is now enabled and will report touches.");
+  }
+
+  zforce.DestroyMessage(msg);
+}
+
+void loop()
+{
+  Message* touch = zforce.GetMessage();
+
+  if (touch != NULL)
+  {
+    if (touch->type == MessageType::TOUCHTYPE)
+    {
+      uint16_t x = ((TouchMessage*)touch)->touchData[0].x;
+      uint16_t y = ((TouchMessage*)touch)->touchData[0].y;
+      uint8_t event = ((TouchMessage*)touch)->touchData[0].event;
+
+      loopMouse(x, y, event);
+      loopKeyboard(x, y, event);
+    }
+    zforce.DestroyMessage(touch);
+  }
+}
```

Comment:
I suggest the following changes

```
void loop()
{
  Message* msg = zforce.GetMessage();

  if (msg != NULL)
  {
    if (msg->type == MessageType::TOUCHTYPE)
    {
      auto touch = static_cast<TouchMessage*>(msg);

      loopMouse(touch->touchData[0].x, touch->touchData[0].y, touch->touchData[0].event);
      loopKeyboard(touch->touchData[0].x, touch->touchData[0].y, touch->touchData[0].event);
    }
    zforce.DestroyMessage(msg);
  }
}
```
alt: 

```
void loop()
{
  Message* msg = zforce.GetMessage();

  if (msg != NULL)
  {
    if (msg->type == MessageType::TOUCHTYPE)
    {
      loopMouse(static_cast<TouchMessage*>(msg));
      loopKeyboard(static_cast<TouchMessage*>(msg));
    }
    zforce.DestroyMessage(msg);
  }
}
```
The code is slightly more optimized as we skip the copying and creation of two new variables.

Reply:
3 new variables *